googlemock/include/gmock/gmock-spec-builders.h

```diff
@@ -1471,7 +1471,14 @@ class FunctionMockerBase : public UntypedFunctionMockerBase {
   virtual ~FunctionMockerBase()
         GTEST_LOCK_EXCLUDED_(g_gmock_mutex) {
     MutexLock l(&g_gmock_mutex);
-    VerifyAndClearExpectationsLocked();
+   
+#if __cplusplus < 201703L
+    const bool inside_uncaught_exception = std::uncaught_exception();
+#else
+    const bool inside_uncaught_exception = std::uncaught_exceptions() > 0;
```

Comment:
This isn't really the intended use of std::uncaught_exceptions(). To use std::uncaught_exceptions() appropriately, you should capture the number of uncaught exceptions on object construction and check again at destruction. If the number at destruction is different (it should always differ by 0 or 1) then you know you can't throw.
See [the proposal](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2014/n4152.pdf).

Also worth mentioning, in the proposal is implementations for std::uncaught_exceptions() for older versions of MSVC and gcc/clang. Perhaps googletest should attempt to use those instead of defaulting to std::uncaught_exception.
See [this github repo](https://github.com/evgeny-panasyuk/stack_unwinding)